libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -132,7 +133,16 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 	string functionName = m_context.functionName(_function);
 	return m_context.functionCollector().createFunction(functionName, [&]() {
 		Whiskers t(R"(
-			function <functionName>(<params>) <returns> {
+			function <functionName>(<params><?isConstructor><comma><additionalParams></isConstructor>) <returns> {
```

Comment:
This looks really complicated. Is there a way to separate the constructor-case and the regular function case?

Reply:
I did refactor it a bit, hope it looks better now. If not, I can completely separate two cases